### PR TITLE
feat(stark-core): add Stark logo to Showcase and in AppLogo component

### PR DESCRIPTION
**Pull request by Mallikki**
ISSUES CLOSED: #738 #136 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #738 #136 

## What is the new behavior?
We added Stark Logo in the project

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+<p align="center">
+  <img src="/assets/images/logo/stark_logo_black.svg" alt="Stark : a powerful front-end framework" width="226" height="50">
```
Comment (christophercr) [inline README.md]: Good news and bad new for this: The logo is shown but... the size is wrong and the font is not the same as in the showcase :s.. 

logo on readme

I guess the issue about the fonts is because the browser doen't know about the font embedded in the SVG... so I think we should adapt the `font-family` of the SVG to use a fallback, which is the same that we also define as`font-family` in the SCSS of Stark.

File: assets/images/logo/stark_logo_black.svg [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0"?>
+<svg width="600" height="200" xmlns="http://www.w3.org/2000/svg" xmlns:svg="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" preserveAspectRatio="xMidYMid meet">
+ <defs>
+  <path d="m326.42,28.24l0,0l-54.41,18.35l8.3,68.05l46.11,24.16l0,0l0,0l46.11,-24.16l8.3,-68.05l-54.41,-18.35l0,0z" id="c15D6WY68D"/>
+  <path d="m326.42,40.51l0,-0.06l0,56l0,0l0,42.35l0,0l46.11,-24.16l8.3,-68.05l-54.41,-18.35l0,0l0,12.27z" id="arvO7VWHV"/>
+  <path d="m347.69,112.6l-6.83,-16.15l-28.87,0l-6.84,16.15l-12.68,0l34.01,-72.15l33.9,72.15l-12.69,0zm-11.27,-26.1l-9.94,-22.61l-9.93,22.61l19.87,0z" id="c7IiT8mTQg"/>
+  <text font-family="Open Sans" font-size="103" font-style="normal" font-weight="normal" id="lAM8KLQl" transform="matrix(1,0,0,1,41.49925519265037,-41.767194972780885) " x="116.46" xml:space="preserve" y="53.27">
```
Comment (christophercr) [inline assets/images/logo/stark_logo_black.svg]: This property is the one I talk about in my previous remark

File: assets/images/logo/stark_logo_black.svg [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+   <tspan dy="0em" id="svg_1" text-anchor="start" x="116.46">ST</tspan>
+  </text>
+  <style>@font-face {
+    font-family: &quot;Open Sans&quot;;
```
Comment (christophercr) [inline assets/images/logo/stark_logo_black.svg]: And also this property is the one I talk about in my previous remark

File: packages/stark-ui/assets/styles/_header.scss [hunk 4]
```diff
@@ -65,21 +65,6 @@
         }
       }
     }
-
-    .stark-app-bar-page-title {
```
Comment (christophercr) [inline packages/stark-ui/assets/styles/_header.scss]: We cannot remove this rules in `stark-ui`. This is needed for any app (maybe not in our showcase) to display the title of the application which is not necessary the same as the logo.

File: packages/stark-ui/assets/styles/_header.scss [hunk 5]
```diff
@@ -103,6 +88,8 @@
         display: flex;
         justify-content: center;
         width: 34%;
+        align-self: center;
+        margin-bottom: 15px;
```
Comment (christophercr) [inline packages/stark-ui/assets/styles/_header.scss]: The same as for my previous remark, we should not add this rules in `stark-ui`, they should be in the showcase instead...

File: packages/stark-ui/assets/styles/_header.scss [hunk 6]
```diff
@@ -122,14 +109,6 @@
           }
         }
       }
-      .stark-app-bar-page-title {
```
Comment (christophercr) [inline packages/stark-ui/assets/styles/_header.scss]: Again, we cannot remove this rules in `stark-ui`

File: packages/stark-ui/assets/styles/_header.scss [hunk 7]
```diff
@@ -181,3 +160,16 @@
     }
   }
 }
+
+@media #{$mobile-only-query} {
```
Comment (christophercr) [inline packages/stark-ui/assets/styles/_header.scss]: Also for these styles: the same as for my previous remark, we should not add this rules in `stark-ui`, they should be in the showcase instead...

File: packages/stark-ui/src/modules/app-logo/components/app-logo.component.scss [hunk 8]
```diff
@@ -10,25 +10,12 @@
...
   & i {
     display: block;
-    width: 60px;
-    height: 64px;
-    margin-top: 4px;
-    background-image: url(/assets/stark-ui/app-logo/angular.png); /* FIXME: add image for tablet */
+    width: 150px;
+    height: 60px;
+    margin-top: 0;
+    background-image: url(/assets-base/stark_logo_transparent.svg); /* FIXME: add image for tablet */
```
Comment (christophercr) [inline packages/stark-ui/src/modules/app-logo/components/app-logo.component.scss]: I think we can remove the FIXME comment now ;)

File: assets/images/logo/stark_logo_black.svg [hunk 9]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0"?>
```
Comment (christophercr) [inline assets/images/logo/stark_logo_black.svg]: The SVGs of the logos should be placed in `packages/stark-core/assets`... following the same structure as in stark-ui.

Then, you need to add the "assets" entry in the `angular.json` file in `packages/stark-core` to add these logos so they will be copied into the app assets while building the app. See how is done in the `angular.json` file in `packages/stark-ui`

File: showcase/src/assets-base/stark_logo_black.svg [hunk 10]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0"?>
```
Comment (christophercr) [inline showcase/src/assets-base/stark_logo_black.svg]: The logos should not be included in the showcase, they should be part of `stark-core` only... check my other remark

File: showcase/src/assets/images/logo/stark_logo.txt [hunk 11]
```diff
@@ -0,0 +1,30 @@
+Inspiration for Stark logo
```
Comment (christophercr) [inline showcase/src/assets/images/logo/stark_logo.txt]: Also this one, it should not be in the showcase

File: showcase/src/assets/images/logo/stark_logo_black.svg [hunk 12]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0"?>
```
Comment (christophercr) [inline showcase/src/assets/images/logo/stark_logo_black.svg]: The same, these logos should not be in the showcase...

File: starter/src/assets-base/stark_logo.txt [hunk 13]
```diff
@@ -0,0 +1,30 @@
+Inspiration for Stark logo
```
Comment (christophercr) [inline starter/src/assets-base/stark_logo.txt]: The same, these logos should not be in the starter...

File: packages/stark-ui/src/modules/app-logo/components/app-logo.component.scss [hunk 14]
```diff
@@ -10,25 +10,12 @@
...
   & i {
     display: block;
-    width: 60px;
-    height: 64px;
-    margin-top: 4px;
-    background-image: url(/assets/stark-ui/app-logo/angular.png); /* FIXME: add image for tablet */
+    width: 150px;
+    height: 60px;
+    margin-top: 0;
+    background-image: url(/assets-base/stark_logo_transparent.svg);
```
Comment (christophercr) [inline packages/stark-ui/src/modules/app-logo/components/app-logo.component.scss]: This path is wrong, the right path should be the one you define in the `angular.json` file of the stark-core ;)

Basically, you need to add the "assets" entry in the `angular.json` file in `packages/stark-core` to add these logos so they will be copied into the app assets while building the app. See how is done in the `angular.json` file in `packages/stark-ui`

File: showcase/src/app/app.component.scss [hunk 15]
```diff
@@ -24,3 +24,20 @@
...
+@media #{$mobile-only-query} {
+  .stark-app-sidebar {
+    .stark-app-sidenav-menu {
+      .stark-app-logo {
+        & i {
+          display: block;
+          width: 150px;
+          height: 60px;
+          margin-top: 0;
+          background-image: url(/assets/stark-core/logo/stark_logo_black.svg);
```
Comment (christophercr) [inline showcase/src/app/app.component.scss]: Is this working? I don't think that this path is configured anywhere.... the right path should be the one you define in the `angular.json` file of the stark-core ;)

File: showcase/src/app/home/_home.component.scss [hunk 16]
```diff
@@ -1,16 +1,8 @@
...
-  margin-left: 15px;
-  color: mat-color($grey-palette, 100);
-}
-
 //set the header image
 .about-background {
-  background: url("/assets/images/banner/angular-banner.png") no-repeat center center;
-  background-size: cover;
-  height: 170px;
+  background: mat-color($primary-palette, 900) url("/assets/stark-core/logo/stark_logo_transparent.svg") no-repeat center center;
```
Comment (christophercr) [inline showcase/src/app/home/_home.component.scss]: Is this working? I don't think that this path is configured anywhere.... the right path should be the one you define in the `angular.json` file of the stark-core ;)

File: showcase/src/app/home/_home.component.scss [hunk 17]
```diff
@@ -62,6 +55,27 @@
...
+  }
+}
+
+@media #{$mobile-only-query} {
+  .stark-app-bar-page-title {
+    top: 60px;
+    left: 0;
+    position: absolute;
+    max-width: 100%;
+    background-image: url(/assets/stark-core/logo/stark_logo_transparent.svg);
```
Comment (christophercr) [inline showcase/src/app/home/_home.component.scss]: Is this working? I don't think that this path is configured anywhere.... the right path should be the one you define in the `angular.json` file of the stark-core ;)

File: starter/src/app/app.component.html [hunk 18]
```diff
@@ -49,7 +49,6 @@
 						</div>
 
 						<div class="stark-app-bar-page-title">
-							<h1>{{ name }}</h1>
```
Comment (christophercr) [inline starter/src/app/app.component.html]: I guess this line should be kept in the Starter right? Because we will only show our logo in the Showcase... or am I missing something?

File: packages/stark-core/angular.json [hunk 19]
```diff
@@ -12,15 +12,27 @@
...
         "test": {
           "options": {
             "main": "base.spec.ts",
             "outputPath": "dist",
             "tsConfig": "tsconfig.spec.json",
-            "assets": []
+            "assets": [
+              {
+                "glob": "**/*",
+                "input": "../node_modules/@nationalbankbelgium/stark-core/assets",
```
Comment (christophercr) [inline packages/stark-core/angular.json]: Why is this added to the assets? I don't think is needed...

File: README.md [hunk 20]
```diff
@@ -1,3 +1,7 @@
+<p align="center">
+  <img src="/assets/stark-core/logo/stark_logo_black.png" alt="Stark : a powerful front-end framework" width="150" height="60">
```
Comment (christophercr) [inline README.md]: This path is wrong... it is pointing to an "assets" folder in the root, but the image is in this path: "packages/stark-core/assets/logo/stark_logo_black.png"

**Comment by christophercr** [comment]
Can you also remove the entry related to the `node_modules/@nationalbankbelgium/stark-ui/src/modules/app-logo/assets`  from the `angular.json` of Stark-UI?

File: showcase/src/app/app.component.scss (hunk 15 above)
Comment (SuperITMan) [reply showcase/src/app/app.component.scss]: It is working 😊

File: showcase/src/app/home/_home.component.scss (hunk 16 above)
Comment (SuperITMan) [reply showcase/src/app/home/_home.component.scss]: I can confirm, it is working too 😊

File: showcase/src/app/home/_home.component.scss (hunk 17 above)
Comment (SuperITMan) [reply showcase/src/app/home/_home.component.scss]: It is working too
